### PR TITLE
target property in a sender/receiver can be an empty object.

### PR DESCRIPTION
**Pull request by amarzavery**

### CODE REVIEW

**Comment by grs** [comment]
@amarzavery it looks like a new version of @types/node may have broken things? Perhaps it is not compatible with the tsc version?

**Comment by amarzavery** [comment]
Looks like they made things stricter. Was able to repro this on my machine after deleting node_modules and package-lock.json. The above commit should fix this.

File: test/links.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ describe('link fields', function() {
                 context.connection.close();
                 done();
             });
-            var c = container.connect(listener.address());
+            var c = container.connect(listener.address() as any);
```
Comment (amarzavery) [inline test/links.ts]: As per the type definitions, connect takes an object of type ConnectionOptions whereas we are passing an object of type AddressInfo. Casting it to `any` fixes the problems.

**Comment by amarzavery** [comment]
@grs - The CI passed. Can you please review the changes and merge if everything looks good?

**Pull request merged**